nemo/collections/tts/models/t5tts.py

```diff
@@ -0,0 +1,1528 @@
+from nemo.collections.tts.losses.aligner_loss import ForwardSumLoss
+from nemo.collections.tts.models import AudioCodecModel
+from nemo.collections.tts.modules import transformer_2501
+from nemo.collections.tts.parts.utils.helpers import get_mask_from_lengths, plot_alignment_to_numpy
+from nemo.collections.tts.parts.utils.tts_dataset_utils import stack_tensors
+from nemo.core.classes import ModelPT
+from nemo.core.classes.common import PretrainedModelInfo
+from nemo.utils import logging, model_utils
+
+HAVE_WANDB = True
+try:
+    import wandb
+except ModuleNotFoundError:
+    HAVE_WANDB = False
+
+
+def setup_tokenizers(all_tokenizers_config, use_text_conditioning_tokenizer, mode='train'):
+    # Being used in both model and worker_init_fn, so it is defined here
+    # Returns two tokenizers: one for TTS transcript and one for conditioning text (if needed)
+    tokenizers = []
+    tokenizer_names = []
+    for tokenizer_name in all_tokenizers_config:
+        tokenizer_config = all_tokenizers_config[tokenizer_name]
+        if tokenizer_config._target_ == 'AutoTokenizer':
+            tokenizer = AutoTokenizer.from_pretrained(tokenizer_config.pretrained_model)
+        else:
+            text_tokenizer_kwargs = {}
+            if "g2p" in tokenizer_config:
+                text_tokenizer_kwargs["g2p"] = instantiate(tokenizer_config.g2p)
+            tokenizer = instantiate(tokenizer_config, **text_tokenizer_kwargs)
+            if mode == 'test' and hasattr(tokenizer, "set_phone_prob"):
+                tokenizer.set_phone_prob(1.0)
+        tokenizers.append(tokenizer)
+        tokenizer_names.append(tokenizer_name)
+
+    aggregated_tokenizer = AggregatedTTSTokenizer(tokenizers, tokenizer_names)  # TTS Transcript tokenizer
+    text_conditioning_tokenizer = None
+
+    if use_text_conditioning_tokenizer:
+        # TODO: make this configurable
+        # Conditioning text tokenizer
+        text_conditioning_tokenizer = T5Tokenizer.from_pretrained("google-t5/t5-small")
+
+    return aggregated_tokenizer, text_conditioning_tokenizer
+
+
+def worker_init_fn(worker_id):
+    # For mp.set_start_method("spawn", force=True)
+    # The dataset class should be picklable, so we initialize non-picklable objects here
+    logging.info(f"Worker {worker_id} initializing...")
+    worker_info = get_worker_info()
+    dataset = worker_info.dataset  # Get the dataset instance in this worker
+    tokenizer, text_conditioning_tokenizer = setup_tokenizers(
+        dataset.tokenizer_config, dataset.use_text_conditioning_tokenizer, mode=dataset.dataset_type
+    )
+    dataset.text_tokenizer = tokenizer
+    dataset.text_conditioning_tokenizer = text_conditioning_tokenizer
+
+
+class T5TTS_Model(ModelPT):
+    def __init__(self, cfg: DictConfig, trainer: 'Trainer' = None):
+        # Convert to Hydra 1.0 compatible DictConfig
+        cfg = model_utils.convert_model_config_to_dict_config(cfg)
+        cfg = model_utils.maybe_update_config_version(cfg)
```

Comment:
we should remove these two lines to take care of any legacy hydra<1.0 since our recipe starts with `hydra-core>1.3,<=1.3.2`